tests/unit/test_charm.py

```diff
@@ -48,6 +49,9 @@ def _requirer_provide_ingress_requirements(
     if strip_prefix:
         data["strip-prefix"] = "true"
 
+    if redirect_https:
+        data["redirect-https"] = "true"
```

Comment:
[thought] I personally don't like optionally present keys. I'd rather it be there and be 'false' than it maybe being there and it not being there might mean that the interface is obsolete OR that redirect-https is False. 
Still, for backwards compatibility, the provider shouldn't complain if it is missing: that's ok.